python_modules/libraries/dagster-dbt/dagster_dbt/cli/solids.py

```diff
@@ -535,3 +535,188 @@ def dbt_cli_compile(context) -> Dict:
+            default_value=None,
+            is_required=False,
+            description="The selector name to use, as defined in your selectors.yml",
+        ),
+        "state": Field(
+            config=Noneable([str]),
+            default_value=None,
+            is_required=False,
+            description=(
+                "If set, use the given directory as the source for json files to compare with "
+                "this project."
+            ),
+        ),
+        "yield_materializations": Field(
+            config=Bool, is_required=False, default_value=True, description="FIXME"
+        ),
+    },
+    tags={"kind": "dbt"},
+)
+@experimental
+def dbt_cli_docs_generate(context) -> Dict:
+    """This solid executes ``dbt docs generate`` via the dbt CLI."""
+    cli_output = execute_cli(
+        context.solid_config["dbt_executable"],
+        command=("docs", "generate",),
+        flags_dict=passthrough_flags_only(
+            context.solid_config,
+            ("threads", "no-version-check", "models", "exclude", "selector", "state",),
+        ),
+        log=context.log,
+        warn_error=context.solid_config["warn-error"],
+        ignore_handled_error=context.solid_config["ignore_handled_error"],
+    )
+
+    if context.solid_config["yield_materializations"]:
+        yield AssetMaterialization(
+            asset_key="dbt_compile_cli_output",
+            description="Output from the CLI execution of `dbt docs generate`.",
+            metadata_entries=[EventMetadataEntry.json(cli_output, label="CLI Output")],
+        )
+
+    yield Output(cli_output)
+
+
+@solid(
+    description="A solid to invoke dbt seed via CLI.",
+    input_defs=[InputDefinition(name="start_after", dagster_type=Nothing)],
+    output_defs=[OutputDefinition(name="result", dagster_type=Dict)],
+    config_schema={
+        **CLI_CONFIG_SCHEMA,
+        "full-refresh": Field(
+            config=bool,
+            default_value=False,
+            is_required=False,
+            description=("Drop existing seed tables and recreate them."),
+        ),
+        "show": Field(
+            config=bool,
+            default_value=False,
+            is_required=False,
+            description=("Show a sample of the loaded data in the terminal."),
+        ),
+        "threads": Field(
+            config=Noneable(int),
+            default_value=None,
+            is_required=False,
+            description=(
+                "Specify number of threads to use while executing models. Overrides settings "
+                "in profiles.yml."
+            ),
+        ),
+        "no-version-check": Field(
+            config=bool,
+            description=(
+                "Skip the check that dbt's version matches the one specified in the "
+                "dbt_project.yml file ('require-dbt-version')"
+            ),
+            is_required=False,
+            default_value=False,
+        ),
+        "models": Field(
```

Comment:
are these config fields relevant to seed?

Reply:
Running `dbt seed -h` shows me that the `--models` flag is actually named `--select`

```
-s MODELS [MODELS ...], --select MODELS [MODELS ...]
                        Specify the nodes to include.
```

it functions similarly to the `--models` flag, but perhaps it should be renamed to `select` for consistency with the official dbt CLI

Reply:
@mgasner @chenbobby It looks like there are two config options missing currently form the RPC versions of these solids: `select` and `full_refresh`. We actually already have the `seed` and `docs generate` solids in our project and will need them once we migrate over to using the `dagster-dbt` library, so I'm following this PR just to keep a heads up.

Reply:
Thanks for pointing that out

And, as we discussed @dwallace0723 , the dedicated RPC endpoint for `dbt seed` [doesn't appear to support](https://docs.getdbt.com/reference/commands/rpc#run-seeds) these additional config options, but this can be circumvented by using [the "CLI over RPC" endpoint](https://docs.getdbt.com/reference/commands/rpc#running-a-task-with-cli-syntax)

this would improve more than just this dbt RPC solid, i think. we can create another PR for such a change